# Tugas 1 Python Cahya Idznii Igawati

Introduction to Python 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import data 1

In [2]:
df = pd.read_csv('[BRI] Python Assessment Data.csv')
df.head()

No Transaction Number Merchant ID Tipe Transaksi Nama Issuer   Nominal  \
0   35         B005215005  X610005448     Pembayaran        DANA  100000.0   
1   64         B005214931  X610110384     Pembayaran        DANA  150000.0   
2   79         B005214895  X612094437     Pembayaran        DANA   20000.0   
3  117         B005744332  X610110454     Pembayaran        DANA  100000.0   
4  130         B005214792  X610115833     Pembayaran        DANA   50000.0   

      Transaction Date   Status  
0  2022-07-10 23:44:59  success  
1  2022-07-10 23:36:31  success  
2  2022-07-10 23:32:14  success  
3  2022-07-20 23:32:39  success  
4  2022-07-10 23:18:16  success

Indentifikasi Database dan Cleansing Database

In [3]:
#melihat tipe data
df.dtypes

No                      int64
Transaction Number     object
Merchant ID            object
Tipe Transaksi         object
Nama Issuer            object
Nominal               float64
Transaction Date       object
Status                 object
dtype: object

dari hasil data satu didapatkan hasil bahwa variabel nominal mempunyai tipe data float sedangkan untuk lainnya adalah object sehingga tipe data untuk variabel transaction date harus diubah dalam bentuk date

In [4]:
#mengubah Transaction Date menjadi tipe date
df["Transaction Date"] = pd.to_datetime(df["Transaction Date"], format = "%Y-%m-%d %H:%M:%S")
df.head()

No Transaction Number Merchant ID Tipe Transaksi Nama Issuer   Nominal  \
0   35         B005215005  X610005448     Pembayaran        DANA  100000.0   
1   64         B005214931  X610110384     Pembayaran        DANA  150000.0   
2   79         B005214895  X612094437     Pembayaran        DANA   20000.0   
3  117         B005744332  X610110454     Pembayaran        DANA  100000.0   
4  130         B005214792  X610115833     Pembayaran        DANA   50000.0   

     Transaction Date   Status  
0 2022-07-10 23:44:59  success  
1 2022-07-10 23:36:31  success  
2 2022-07-10 23:32:14  success  
3 2022-07-20 23:32:39  success  
4 2022-07-10 23:18:16  success

In [5]:
#menghapus variabel Nomor dan Status
df.drop(columns = ["No", "Status"], inplace = True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11000 entries, 0 to 10999
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Transaction Number  11000 non-null  object        
 1   Merchant ID         10966 non-null  object        
 2   Tipe Transaksi      11000 non-null  object        
 3   Nama Issuer         11000 non-null  object        
 4   Nominal             10989 non-null  float64       
 5   Transaction Date    11000 non-null  datetime64[ns]
dtypes: datetime64[ns](1), float64(1), object(4)
memory usage: 515.8+ KB


In [6]:
#memisahkan tanggal dan waktu
df['Tanggal']= df["Transaction Date"].dt.date
df['Waktu']= df["Transaction Date"].dt.time
df['Jam']= df["Transaction Date"].dt.hour
df.head()

Transaction Number Merchant ID Tipe Transaksi Nama Issuer   Nominal  \
0         B005215005  X610005448     Pembayaran        DANA  100000.0   
1         B005214931  X610110384     Pembayaran        DANA  150000.0   
2         B005214895  X612094437     Pembayaran        DANA   20000.0   
3         B005744332  X610110454     Pembayaran        DANA  100000.0   
4         B005214792  X610115833     Pembayaran        DANA   50000.0   

     Transaction Date     Tanggal     Waktu  Jam  
0 2022-07-10 23:44:59  2022-07-10  23:44:59   23  
1 2022-07-10 23:36:31  2022-07-10  23:36:31   23  
2 2022-07-10 23:32:14  2022-07-10  23:32:14   23  
3 2022-07-20 23:32:39  2022-07-20  23:32:39   23  
4 2022-07-10 23:18:16  2022-07-10  23:18:16   23

Melihat dan menghapus missing value

In [7]:
#menghitung missing value tiap variabel
df.isna().sum()

Transaction Number     0
Merchant ID           34
Tipe Transaksi         0
Nama Issuer            0
Nominal               11
Transaction Date       0
Tanggal                0
Waktu                  0
Jam                    0
dtype: int64

In [8]:
#menghapus data missing
df = df.dropna()

In [9]:
#menghitung missing value tiap variabel
df.isna().sum()

Transaction Number    0
Merchant ID           0
Tipe Transaksi        0
Nama Issuer           0
Nominal               0
Transaction Date      0
Tanggal               0
Waktu                 0
Jam                   0
dtype: int64

sebelum mengolah maka kita harus melakukan cleaning data dimana data tidak boleh terdeteksi duplikat sehingga akan menghapus data duplikat untuk semua variabel yang sama sehingga ddihapuskan duplikat yang sama untuk setiap variabelnya

In [10]:
#menghapus duplikat yang sama setiap variabelnya 
df_new = df.drop_duplicates()
df_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10120 entries, 0 to 10999
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Transaction Number  10120 non-null  object        
 1   Merchant ID         10120 non-null  object        
 2   Tipe Transaksi      10120 non-null  object        
 3   Nama Issuer         10120 non-null  object        
 4   Nominal             10120 non-null  float64       
 5   Transaction Date    10120 non-null  datetime64[ns]
 6   Tanggal             10120 non-null  object        
 7   Waktu               10120 non-null  object        
 8   Jam                 10120 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(6)
memory usage: 790.6+ KB


setelah data yang merupakan duplikat sama untuk semua variabel dihapus untuk duplikatnya, maka selanjutnya adalah kita menghapus data yang memiliki transaction number sama. hal ini adalah untuk data dengan number transaction sama tetapi dalam statusnya pembayaran dan refund sehingga kita harus menghapus kedua transaksi tersebut.

In [11]:
#menghapus duplikat berdasarkan variabel Transaction Number
df = df_new.drop_duplicates(subset=['Transaction Number'], keep=  False  )
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10016 entries, 0 to 10999
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Transaction Number  10016 non-null  object        
 1   Merchant ID         10016 non-null  object        
 2   Tipe Transaksi      10016 non-null  object        
 3   Nama Issuer         10016 non-null  object        
 4   Nominal             10016 non-null  float64       
 5   Transaction Date    10016 non-null  datetime64[ns]
 6   Tanggal             10016 non-null  object        
 7   Waktu               10016 non-null  object        
 8   Jam                 10016 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(6)
memory usage: 782.5+ KB


In [23]:
#melihat proporsi kategori tipe transaksi
df['Tipe Transaksi'].value_counts()

Pembayaran    10000
Refund           16
Name: Tipe Transaksi, dtype: int64

In [24]:
#mengecek apakah terdapat duplicate
df['Transaction Number'].duplicated().any()

False

In [25]:
#mengecek apakah terdapat duplicate untuk semua variabel
df.duplicated().any()

False

terdapat 16 data yang merupakan data refund dan untuk prmbayaran adalah 10000 data

Gabung Dengan Data2 

In [26]:
df2 =pd.read_csv('[BRI] Python Assessment Data 2.csv')
df2.head()

Merchant ID  Merchant Criteria     Merchant Type
0  X610005443  (UMI) Usaha Mikro       Restaurants
1  X610005869  (UMI) Usaha Mikro     Miscellaneous
2  X314022534  (UKE) Usaha Kecil  Food & Beverages
3  X612400459  (UKE) Usaha Kecil     Miscellaneous
4  X614400335  (UKE) Usaha Kecil     Miscellaneous

In [27]:
#menghitung missing value tiap variabel
df2.isna().sum()

Merchant ID          0
Merchant Criteria    0
Merchant Type        0
dtype: int64

tidak terdapat missing value pada data kedua sehingga dilanjutkan digabungkan data pertama dan kedua

In [28]:
#gabung tabel
df_join = df.merge(df2, on='Merchant ID', how='left')
df_join.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10016 entries, 0 to 10015
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Transaction Number  10016 non-null  object        
 1   Merchant ID         10016 non-null  object        
 2   Tipe Transaksi      10016 non-null  object        
 3   Nama Issuer         10016 non-null  object        
 4   Nominal             10016 non-null  float64       
 5   Transaction Date    10016 non-null  datetime64[ns]
 6   Tanggal             10016 non-null  object        
 7   Waktu               10016 non-null  object        
 8   Jam                 10016 non-null  int64         
 9   Merchant Criteria   10008 non-null  object        
 10  Merchant Type       10008 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(8)
memory usage: 939.0+ KB


dilakukan cleansing data yang telah digabungkan dan menghapus missing value nya

In [29]:
#menghapus data missing dan cek missing
df_join_nomissing = df_join.dropna() 
df_join_nomissing.isna().sum()

Transaction Number    0
Merchant ID           0
Tipe Transaksi        0
Nama Issuer           0
Nominal               0
Transaction Date      0
Tanggal               0
Waktu                 0
Jam                   0
Merchant Criteria     0
Merchant Type         0
dtype: int64

In [30]:
df_join_nomissing.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10008 entries, 0 to 10015
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Transaction Number  10008 non-null  object        
 1   Merchant ID         10008 non-null  object        
 2   Tipe Transaksi      10008 non-null  object        
 3   Nama Issuer         10008 non-null  object        
 4   Nominal             10008 non-null  float64       
 5   Transaction Date    10008 non-null  datetime64[ns]
 6   Tanggal             10008 non-null  object        
 7   Waktu               10008 non-null  object        
 8   Jam                 10008 non-null  int64         
 9   Merchant Criteria   10008 non-null  object        
 10  Merchant Type       10008 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(8)
memory usage: 938.2+ KB


Mengambil data yang nominal transaksi dibawah 10.000.000

In [31]:
#di filter nominal<10.000.000
df_filter = df_join_nomissing[df_join_nomissing["Nominal"]<=10000000]
df_filter.head()

Transaction Number Merchant ID Tipe Transaksi Nama Issuer   Nominal  \
0         B005215005  X610005448     Pembayaran        DANA  100000.0   
1         B005214931  X610110384     Pembayaran        DANA  150000.0   
2         B005214895  X612094437     Pembayaran        DANA   20000.0   
3         B005744332  X610110454     Pembayaran        DANA  100000.0   
4         B005214792  X610115833     Pembayaran        DANA   50000.0   

     Transaction Date     Tanggal     Waktu  Jam  Merchant Criteria  \
0 2022-07-10 23:44:59  2022-07-10  23:44:59   23  (UMI) Usaha Mikro   
1 2022-07-10 23:36:31  2022-07-10  23:36:31   23  (UMI) Usaha Mikro   
2 2022-07-10 23:32:14  2022-07-10  23:32:14   23  (UMI) Usaha Mikro   
3 2022-07-20 23:32:39  2022-07-20  23:32:39   23  (UMI) Usaha Mikro   
4 2022-07-10 23:18:16  2022-07-10  23:18:16   23  (UMI) Usaha Mikro   

                Merchant Type  
0                 Restaurants  
1                 Restaurants  
2  Supermarket/Grocery Stores  
3                 Restaurants  
4               Miscellaneous

In [32]:
df_filter.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10008 entries, 0 to 10015
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Transaction Number  10008 non-null  object        
 1   Merchant ID         10008 non-null  object        
 2   Tipe Transaksi      10008 non-null  object        
 3   Nama Issuer         10008 non-null  object        
 4   Nominal             10008 non-null  float64       
 5   Transaction Date    10008 non-null  datetime64[ns]
 6   Tanggal             10008 non-null  object        
 7   Waktu               10008 non-null  object        
 8   Jam                 10008 non-null  int64         
 9   Merchant Criteria   10008 non-null  object        
 10  Merchant Type       10008 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(8)
memory usage: 938.2+ KB


In [33]:
#melihat proporsi kategori tipe transaksi
df_filter['Tipe Transaksi'].value_counts()

Pembayaran    9992
Refund          16
Name: Tipe Transaksi, dtype: int64

jumlah data menjadi 9992 karena dari 10.000 data terdapat data yang memiliki Merchanr Kriteria Kosong sehingga dihapuskan untuk analisis kali ini

# Tugas Pivot Tabel 

In [23]:
dfpivot1 = df_filter.pivot_table(values = "Nominal", index= "Merchant ID", aggfunc='sum').reset_index().sort_values(by='Nominal', ascending = False)
dfpivot1.head()

Merchant ID     Nominal
2377  X612500042  22761000.0
876   X605045913  19389500.0
1971  X610153754  12487932.0
1923  X610110452  10243900.0
1972  X610153755   9295888.0

In [34]:
df_pivot2 = df_filter.pivot_table( index= ['Merchant ID','Nama Issuer'], aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'})
df_pivot2

Nominal  Transaction Number
Merchant ID Nama Issuer                               
X1010172    BNI           226600.0                   1
X11017480   MANDIRI        80000.0                   1
X12000017   BNI                1.0                   1
X12126900   BNI          2878520.0                   1
X12208583   BCA            99000.0                   1
...                            ...                 ...
X618500133  BNI             1506.0                   8
X618500135  BNI              405.0                   9
X618500136  BNI               13.0                  13
X6990048    BNI           242000.0                   1
X8064420    BNI           156150.0                   1

[4783 rows x 2 columns]

Nomor 1 Melakukan Pivot dengan Analisi berdasarkan waktu dari database 

In [36]:
# dilakukan sort untuk 'Transaction Date'
df_final = df_filter.sort_values(
            by = 'Transaction Date')

In [37]:
df_final.head()

Transaction Number Merchant ID Tipe Transaksi Nama Issuer   Nominal  \
10015         B004561428  X610153755     Pembayaran        DANA  100000.0   
10014         B004561651  X610006258     Pembayaran        DANA   30000.0   
10013         B004561820  X610153755     Pembayaran        DANA   50000.0   
10012         B004561975  X610153453     Pembayaran         OVO   49412.0   
10011         B004562347  X612094437     Pembayaran        DANA   68000.0   

         Transaction Date     Tanggal     Waktu  Jam  Merchant Criteria  \
10015 2022-07-01 00:13:49  2022-07-01  00:13:49    0  (UMI) Usaha Mikro   
10014 2022-07-01 00:43:01  2022-07-01  00:43:01    0  (UMI) Usaha Mikro   
10013 2022-07-01 01:12:04  2022-07-01  01:12:04    1  (UMI) Usaha Mikro   
10012 2022-07-01 01:44:30  2022-07-01  01:44:30    1  (UMI) Usaha Mikro   
10011 2022-07-01 03:26:26  2022-07-01  03:26:26    3  (UMI) Usaha Mikro   

                    Merchant Type  
10015                 Restaurants  
10014                 Restaurants  
10013                 Restaurants  
10012    Automobiles And Vehicles  
10011  Supermarket/Grocery Stores

In [39]:
#pembagian kategori jam 

#Midnight: 0.00 – 4.00
#Morning: 4.00 – 11.00
#Noon: 11.00  – 13.00 
#Afternoon: 13.00 – 17.00 
#Night: 18.00  – 24.00 

In [59]:
Condition = [
    (df_final['Jam'] >= 0) & (df_final['Jam'] < 4),
    (df_final['Jam'] < 11) & (df_final['Jam'] >= 4),
    (df_final['Jam'] < 13) & (df_final['Jam'] >= 11),
    (df_final['Jam'] < 17) & (df_final['Jam'] >= 13),
    (df_final['Jam'] <= 24) & (df_final['Jam'] >= 17)
]

Categori = ['Midnight','Morning','Noon','Afternoon','Night']

df_final['Kategori Jam']= np.select(Condition, Categori)

In [60]:
df_final.head()

Transaction Number Merchant ID Tipe Transaksi Nama Issuer   Nominal  \
10015         B004561428  X610153755     Pembayaran        DANA  100000.0   
10014         B004561651  X610006258     Pembayaran        DANA   30000.0   
10013         B004561820  X610153755     Pembayaran        DANA   50000.0   
10012         B004561975  X610153453     Pembayaran         OVO   49412.0   
10011         B004562347  X612094437     Pembayaran        DANA   68000.0   

         Transaction Date     Tanggal     Waktu  Jam  Merchant Criteria  \
10015 2022-07-01 00:13:49  2022-07-01  00:13:49    0  (UMI) Usaha Mikro   
10014 2022-07-01 00:43:01  2022-07-01  00:43:01    0  (UMI) Usaha Mikro   
10013 2022-07-01 01:12:04  2022-07-01  01:12:04    1  (UMI) Usaha Mikro   
10012 2022-07-01 01:44:30  2022-07-01  01:44:30    1  (UMI) Usaha Mikro   
10011 2022-07-01 03:26:26  2022-07-01  03:26:26    3  (UMI) Usaha Mikro   

                    Merchant Type Kategori Jam  
10015                 Restaurants     Midnight  
10014                 Restaurants     Midnight  
10013                 Restaurants     Midnight  
10012    Automobiles And Vehicles     Midnight  
10011  Supermarket/Grocery Stores     Midnight

In [61]:
df_pivot2 = df_final.pivot_table( index= ['Merchant ID','Nama Issuer'], aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'})
df_pivot2

Nominal  Transaction Number
Merchant ID Nama Issuer                               
X1010172    BNI           226600.0                   1
X11017480   MANDIRI        80000.0                   1
X12000017   BNI                1.0                   1
X12126900   BNI          2878520.0                   1
X12208583   BCA            99000.0                   1
...                            ...                 ...
X618500133  BNI             1506.0                   8
X618500135  BNI              405.0                   9
X618500136  BNI               13.0                  13
X6990048    BNI           242000.0                   1
X8064420    BNI           156150.0                   1

[4783 rows x 2 columns]

In [74]:
df_pivot3 = df_final.pivot_table( index= ['Nama Issuer','Merchant ID',], aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'}).sort_values(by='Nominal', ascending = False)
df_pivot3

Nominal  Transaction Number
Nama Issuer Merchant ID                               
BCA         X605045913   9134500.0                   3
DANA        X610153755   7226888.0                  91
BCA         X610153754   7000000.0                   1
            X610117010   6606500.0                   2
DANA        X612094437   6488215.0                  88
...                            ...                 ...
SHOPEEPAY   X214901026         1.0                   1
            X214901033         1.0                   1
            X217199991         1.0                   1
            X217200666         1.0                   1
BNI         X611500116         1.0                   1

[4783 rows x 2 columns]

In [92]:
df_pivot3 = df_final.pivot_table( index= ['Nama Issuer','Merchant ID',], aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'}).sort_values(by='Transaction Number', ascending = False)
df_pivot3

Nominal  Transaction Number
Nama Issuer Merchant ID                               
BNI         X610002918   4354102.0                 514
OVO         X601500090    608000.0                 128
DANA        X610153755   7226888.0                  91
            X612094437   6488215.0                  88
BNI         X601500166       162.0                  87
...                            ...                 ...
            X610500648    130100.0                   1
            X610501124     34000.0                   1
            X610501132     89000.0                   1
            X611001498     70400.0                   1
SPIN        X610500605     53500.0                   1

[4783 rows x 2 columns]

In [73]:
df_pivot4 = df_final.pivot_table( index= ['Nama Issuer','Merchant ID','Tanggal','Jam'], aggfunc={'Nominal':'sum', 'Nominal' : 'count'})
df_pivot4

Nominal
Nama Issuer Merchant ID Tanggal    Jam         
ARTHA GRAHA X608011397  2022-07-17 15         1
ASTRAPAY    X607102105  2022-07-02 11         1
            X612094437  2022-07-23 12         1
            X612500042  2022-07-07 17         1
Allo Bank   X602194737  2022-07-07 15         1
...                                         ...
SINARMAS    X607081667  2022-07-25 19         1
            X607102817  2022-07-17 8          1
            X609162039  2022-07-13 18         1
            X616500024  2022-07-30 16         1
SPIN        X610500605  2022-07-28 7          1

[9222 rows x 1 columns]

In [65]:
df_pivot5 = df_final.pivot_table( index= ['Nama Issuer','Merchant ID','Kategori Jam','Jam'], aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'})
df_pivot5

Nominal  Transaction Number
Nama Issuer Merchant ID Kategori Jam Jam                              
ARTHA GRAHA X608011397  Afternoon    15   136400.0                   1
ASTRAPAY    X607102105  Noon         11    60000.0                   1
            X612094437  Noon         12   100000.0                   1
            X612500042  Night        17    85000.0                   1
Allo Bank   X602194737  Afternoon    15   250000.0                   1
...                                            ...                 ...
SINARMAS    X607081667  Night        19    28125.0                   1
            X607102817  Morning      8     18000.0                   1
            X609162039  Night        18    91000.0                   1
            X616500024  Afternoon    16    57500.0                   1
SPIN        X610500605  Morning      7     53500.0                   1

[7425 rows x 2 columns]

In [66]:
df_final['Kategori Jam'].value_counts()

Night        3204
Afternoon    3011
Morning      1904
Noon         1651
Midnight      238
Name: Kategori Jam, dtype: int64

In [70]:
df_pivot5 = df_final.pivot_table( index= ['Kategori Jam','Merchant Type'], aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'}).sort_values(by='Nominal', ascending = False)
df_pivot5

Nominal  Transaction Number
Kategori Jam Merchant Type                                     
Night        Restaurants        164765573.0                1714
Afternoon    Restaurants        109851755.0                1316
Noon         Restaurants         54086764.0                 775
Night        Miscellaneous       49297828.0                 600
Morning      Restaurants         41451377.0                 543
...                                     ...                 ...
Noon         OTHERS                 37073.0                 394
             Electronic             30000.0                   1
Afternoon    Travel                 26000.0                   2
Midnight     Food & Beverages       15000.0                   1
Afternoon    Watches & Jewelry          3.0                   1

[96 rows x 2 columns]

Nomor 2 Melakukan Pivot dengan Analisi kedua

In [84]:
df_pivot6 = df_final.pivot_table( index= 'Merchant ID', columns= 'Nama Issuer', aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'})
df_pivot6 = df_pivot6.fillna(0)
df_pivot6

Nominal                                             \
Nama Issuer ARTHA GRAHA ASTRAPAY Allo Bank BANK KEB HANA BANK MEGA   
Merchant ID                                                          
X1010172            0.0      0.0       0.0           0.0       0.0   
X11017480           0.0      0.0       0.0           0.0       0.0   
X12000017           0.0      0.0       0.0           0.0       0.0   
X12126900           0.0      0.0       0.0           0.0       0.0   
X12208583           0.0      0.0       0.0           0.0       0.0   
...                 ...      ...       ...           ...       ...   
X618500133          0.0      0.0       0.0           0.0       0.0   
X618500135          0.0      0.0       0.0           0.0       0.0   
X618500136          0.0      0.0       0.0           0.0       0.0   
X6990048            0.0      0.0       0.0           0.0       0.0   
X8064420            0.0      0.0       0.0           0.0       0.0   

                                                                     \
Nama Issuer BANK NATIONAL NOBU BANK QNB INDONESIA BANK UOB      BCA   
Merchant ID                                                           
X1010172                   0.0                0.0      0.0      0.0   
X11017480                  0.0                0.0      0.0      0.0   
X12000017                  0.0                0.0      0.0      0.0   
X12126900                  0.0                0.0      0.0      0.0   
X12208583                  0.0                0.0      0.0  99000.0   
...                        ...                ...      ...      ...   
X618500133                 0.0                0.0      0.0      0.0   
X618500135                 0.0                0.0      0.0      0.0   
X618500136                 0.0                0.0      0.0      0.0   
X6990048                   0.0                0.0      0.0      0.0   
X8064420                   0.0                0.0      0.0      0.0   

                        ... Transaction Number                             \
Nama Issuer        BNI  ...          OCBC NISP  OVO PERMATA POS Indonesia   
Merchant ID             ...                                                 
X1010172      226600.0  ...                0.0  0.0     0.0           0.0   
X11017480          0.0  ...                0.0  0.0     0.0           0.0   
X12000017          1.0  ...                0.0  0.0     0.0           0.0   
X12126900    2878520.0  ...                0.0  0.0     0.0           0.0   
X12208583     134000.0  ...                0.0  0.0     0.0           0.0   
...                ...  ...                ...  ...     ...           ...   
X618500133      1506.0  ...                0.0  0.0     0.0           0.0   
X618500135       405.0  ...                0.0  0.0     0.0           0.0   
X618500136        13.0  ...                0.0  0.0     0.0           0.0   
X6990048      242000.0  ...                0.0  0.0     0.0           0.0   
X8064420      156150.0  ...                0.0  0.0     0.0           0.0   

                                                               \
Nama Issuer PT Bank BPD Kalimantan Timur dan Kalimantan Utara   
Merchant ID                                                     
X1010172                                                  0.0   
X11017480                                                 0.0   
X12000017                                                 0.0   
X12126900                                                 0.0   
X12208583                                                 0.0   
...                                                       ...   
X618500133                                                0.0   
X618500135                                                0.0   
X618500136                                                0.0   
X6990048                                                  0.0   
X8064420                                                  0.0   

                                                  

In [90]:
df_pivot7 = df_final.pivot_table( index= 'Nama Issuer', aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'}).astype(int).sort_values(by='Nominal', ascending = False)
df_pivot7 = df_pivot7.fillna(0)
df_pivot7.head()

Nominal  Transaction Number
Nama Issuer                               
BCA          287559019                1870
BNI          178234692                3982
MANDIRI      119584576                 703
DANA          74358733                1148
BRI           68306929                 374

In [91]:
df_pivot7 = df_final.pivot_table( index= 'Nama Issuer', aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'}).astype(int).sort_values(by='Transaction Number', ascending = False)
df_pivot7 = df_pivot7.fillna(0)
df_pivot7.head()

Nominal  Transaction Number
Nama Issuer                               
BNI          178234692                3982
BCA          287559019                1870
DANA          74358733                1148
MANDIRI      119584576                 703
OVO           31867860                 536

# DESKRIPSI PENGOLAHAN DATA

1. Variabel yang ada digunkan untuk apa ?
Dari tabel pivot diatas variabel yang digunakan berguna untuk beberpa hal yang dapat menggambarkan keadaan dari merchant dan issuer.

- Variabel Merchant Id adalah Kode Merchant dimana menandakan untuk setiap Merchat tergolong dalam kategori kreiteria dan merchant type
-  Variabel Nama Issuer adalah Variabel yang digunakan untuk melihat Isuuer yang ada siapa yang akan menjadi top untuk nominal dan frekuensi transaksi
- Jam ini adalah variabel date yang digunakan untuk melihat trend waktu disetiap merchat atau nama issuer sering melakukan transaksi pada jam keberapa. 
- Tanggal adalah variabel yang dapat digunakan untuk dilakukan analisis timeseries seingga adapat dikelompokan setiap issuer dan dapat dilakukan analisis prediktif
 

2. Informasi yang didapatkan itu apa saja?

- Dalam dataframe pivot 3 dapat diketahui bahwa untuk nama isuuer dengan nominal terbesar adalah BCA dengan nomor Merchant ID X605045913 dengan total nominal Rp. 9.134.500,- dan jumlah traksasi adalah 3 kali transaksi sehingga rata-ratanya adalah Rp 3.044.833,- dalam sekali transaksi
- Dalam dataframe pivot 3 dapat diketahui bahwa untuk frekuensi transaksi terbesar adalah BNI dengan nomor Merchat ID X610002918 dengan total transaksi sebanyak 514 Transaksi.
- Dalam Kategori jam saya mengelompokan mejadi 5 zona kategori dimana kategori tersebut memiliki hasil untuk database tersebut adalah Night 3204 data , Afternoon 3011 data, Morning  1904 data, Noon 1651 data, dan Midnight 238 data
- Dalam Pivot 5 dapat diketahui bajwa Merchat Type paling banyak untuk total nominal dan frekuensi transasksi adalah Restaurants. 
- Merchant Type Restaurants Paling banyak melakukan transaksi pada Malam dan Afternoon
- Untuk Merchant ID X12208583 lebih banyak transaksi di Perbankan hal ini dapat menjadikan evaluasi untuk issuer lain untuk merchat tersebut diberikan promo agar terdapat nilai transaksinya
- Dari database uamh ada Nominal transaksi terbesar adalah BCA, BNI, Mandiri, Dana, dan urutan ke 5 adalah BRI 
- Dari segi frekuensi transaksi dapat diketahui issuer terbanyak frekuensi transaksi adalah BNI, BCA, DANA, Mandiri dan OVO. Hal ini BRI masih belum termasuk 5 issuer yang memiliki frekuensi transaksi terbanyak sehingga perlu ditingkatkan dengan cara penambahan promo atapun sebagainya


3. Informasi apa yang dapat ditambahkan dalam analisis yang telah dilakukan ?
- Dapat dilakukan penambahan variabel lokasi transaksi sehingga dapat terbentuk analisis deskriptif untuk melihat setiap clustering daerah akan seperti apa karakteristiknya dan juga dapat dilakukan analisis pengelompokan lebih lanjut
- Variabel yang dapat ditambahkan selanjutnya adalah karakteristik untuk yang memlakukan transksinya disini kita dapat bertujuan adaalah untuk melihat permodelan dan prediktif
- variabel selanjutnya adalah memabahkan beberapa kategori untuk disetiap issuer dan juga merchant type hal ini adalah bertujuan memudahkan membaca deskriptif secara analisis dan juga analisis lanjutan

Tambahan analisis

In [98]:
df_pivot8 = df_final.pivot_table( index= 'Merchant Criteria', aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'}).astype(int).sort_values(by='Nominal', ascending = False)
df_pivot8 = df_pivot8.fillna(0)
df_pivot8.head()

Nominal  Transaction Number
Merchant Criteria                                  
(UMI) Usaha Mikro     330677876                5014
(UBE) Usaha Besar     253505620                2488
(UKE) Usaha Kecil     179669798                1508
(UME) Usaha Menengah  115794989                 985
(URE) Usaha Reguler     1575924                  13

Dari data diatas dapat diketahui nominal terbesar adalah pada segmentasi pacar merchant criteria Usaha Mikro (UMI) dengan jumlah transaksi adalah sebanayak 5014 transaksi

In [100]:
df_pivot9 = df_final.pivot_table( index= 'Merchant Type', aggfunc={'Nominal':'sum', 'Transaction Number' : 'count'}).astype(int).sort_values(by='Nominal', ascending = False)
df_pivot9 = df_pivot8.fillna(0)
df_pivot9.head()

Nominal  Transaction Number
Merchant Type                                  
Restaurants       383628253                4467
Miscellaneous     128748722                1485
Health Service     84006255                 411
Food & Beverages   51400183                 461
Clothing           34124876                 159

Dari pivot diatas dapat diketahui bahwa untuk nominal terbanyak dan juga transaksi terbanyak adalah pada merchant type restoran